pkg/cmd/spks.go

```diff
@@ -0,0 +1,215 @@
+			&cli.StringFlag{Name: "odoo-oauth-client-secret", Usage: "Client secret of the oauth client to interact with Odoo metered billing API",
+				EnvVars: []string{"ODOO_OAUTH_CLIENT_SECRET"}, Destination: &odooClientSecret, Required: true, DefaultText: defaultTextForRequiredFlags},
+			&cli.StringFlag{Name: "sales-order", Usage: "Sales order for APPUiO Managed clusters",
+				EnvVars: []string{"SALES_ORDER"}, Destination: &salesOrder, Required: false, DefaultText: defaultTextForOptionalFlags, Value: "S10121"},
+			&cli.StringFlag{Name: "prometheus-url", Usage: "URL of the Prometheus API",
+				EnvVars: []string{"PROMETHEUS_URL"}, Destination: &prometheusURL, Required: true, DefaultText: defaultTextForRequiredFlags, Value: "http://prometheus-monitoring-application.monitoring-application.svc.cluster.local:9090"},
+			&cli.StringFlag{Name: "unit-id", Usage: "Unit ID for the consumed units",
+				EnvVars: []string{"UNIT_ID"}, Destination: &UnitID, Required: true, DefaultText: defaultTextForRequiredFlags, Value: "uom_uom_68_b1811ca1"},
+		},
+		Action: func(c *cli.Context) error {
+			logger := log.Logger(c.Context)
+			logger.Info("starting spks data collector")
+
+			logger.Info("Getting specific metric from thanos")
+
+			client, err := api.NewClient(api.Config{
+				Address: prometheusURL,
+			})
+			if err != nil {
+				logger.Error(err, "Error creating Prometheus client")
+				return err
+			}
+
+			v1api := v1.NewAPI(client)
+			ctx, cancel := context.WithTimeout(context.Background(), 60*time.Second)
+			defer cancel()
+			// send this value to odoo
+
+			mariadbStandard, err := QueryPrometheus(ctx, v1api, prometheusQueryArr[0], logger)
+			if err != nil {
+				logger.Error(err, "Error querying Prometheus")
+				return err
+			}
+			mariadbPremium, err := QueryPrometheus(ctx, v1api, prometheusQueryArr[1], logger)
+			if err != nil {
+				logger.Error(err, "Error querying Prometheus")
+				return err
+			}
+			redisStandard, err := QueryPrometheus(ctx, v1api, prometheusQueryArr[2], logger)
+			if err != nil {
+				logger.Error(err, "Error querying Prometheus")
+				return err
+			}
+			redisPremium, err := QueryPrometheus(ctx, v1api, prometheusQueryArr[3], logger)
+			if err != nil {
+				logger.Error(err, "Error querying Prometheus")
+				return err
+			}
+
+			odooClient := odoo.NewOdooAPIClient(c.Context, odooURL, odooOauthTokenURL, odooClientId, odooClientSecret, logger)
+			location, err := time.LoadLocation("Europe/Zurich")
+			if err != nil {
+				return fmt.Errorf("load loaction: %w", err)
+			}
+
+			from := time.Now().In(location).Add(-(time.Hour * 24)).UTC()
+			to := time.Now().In(location).UTC()
+
+			billingRecords := []odoo.OdooMeteredBillingRecord{
+				{
+					ProductID:            "appcat-spks-mariadb-standard",
+					InstanceID:           "mariadb-standard",
+					ItemDescription:      "appcat-spks-mariadb-standard",
+					ItemGroupDescription: "SPKS",
```

Comment:
This doesn't make a lot of sense. These two parameters end up in the invoice and should be descriptive to the consumer of the invoice (see https://docs.central.vshn.ch/metered-billing-data-ingestion.html#_json_structure). See example JSON in SPKS-863 for the "ItemDescription". "ItemGroupDescription" should be removed, it's optional anyway.